src/haz3lcore/lang/term/Typ.re

```diff
@@ -562,7 +562,7 @@ let rec get_sum_constructors = (ctx: Ctx.t, ty: t): option(sum_map) => {
        the below code will be incorrect! */
     let ty =
       switch (ty |> term_of) {
-      | Rec({term: Var(x), _}, ty_body) =>
+      | Rec({term: Var(x), _}, _ty_body) =>
```

Comment:
@Negabinary this is the case that I caught it for. Lmk if this is actually a bug.